### PR TITLE
Get Cryotank indexing ported

### PR DESCRIPTION
**Pull request by jnwatson**
Cryotank unit tests now pass.

### CODE REVIEW

File: synapse/cryotank.py [hunk 1]
```diff
@@ -1256,18 +1100,18 @@ def _iterrows(self, prop, valu, exact=False):
                 if not curs.next():
                     return
 
-    def queryNormValu(self, prop, valu=None, exact=False):
+    def queryNormValu(self, prop: str, valu: Optional[Union[int, str]] = None, exact=False):
```
Comment (vertexmc) [inline synapse/cryotank.py]: If we're adding type annotations to these methods, `exact` should get one too

File: synapse/cryotank.py [hunk 2]
```diff
@@ -1277,18 +1121,19 @@ def queryNormValu(self, prop, valu=None, exact=False):
...
-            sorted by prop will be returned.  It will be considered a prefix if exact is False.
-            exact (bool): Indicates that the result must match exactly.  Conversely, if False, indicates a prefix match.
+            prop:  The name of the indexed property
+            valu:  The normalized value.  If not present, all records with prop present, sorted by prop will be
+                returned.  It will be considered a prefix if exact is False.
+            exact: Indicates that the result must match exactly.  Conversely, if False, indicates a prefix match.
 
         Returns:
-            Iterable[Tuple[int, Dict[str, Union[str, int]]]]: A generator of offset, dictionary tuples
+            : A generator of offset, dictionary tuples
```
Comment (vertexmc) [inline synapse/cryotank.py]: Should we remove the `: ` there?

File: synapse/cryotank.py [hunk 3]
```diff
@@ -1277,18 +1121,19 @@ def queryNormValu(self, prop, valu=None, exact=False):
                 raise s_exc.CorruptDatabase('Missing normalized record')  # pragma: no cover
             yield offset, s_msgpack.un(rv)
 
-    def queryNormRecords(self, prop, valu=None, exact=False):
+    def queryNormRecords(self, prop: str, valu: Optional[Union[int, str]] = None, exact=False) -> \
```
Comment (vertexmc) [inline synapse/cryotank.py]: Same `exact` comment as above

File: synapse/lib/cell.py [hunk 4]
```diff
@@ -174,7 +178,7 @@ class Cell(s_eventbus.EventBus, s_telepath.Aware):
     '''
     cellapi = CellApi
 
-    confdefs = ()
+    confdefs: Tuple[Any, ...] = ()
```
Comment (vertexmc) [inline synapse/lib/cell.py]: Awesome. I wasn't sure how to do this, I can clean up some of the cmdr typing warnings.

File: synapse/lib/cell.py [hunk 5]
```diff
@@ -1,12 +1,16 @@
...
 import logging
 import threading
+from typing import Any, Tuple
 
 import synapse.exc as s_exc
 import synapse.common as s_common
 import synapse.eventbus as s_eventbus
 import synapse.telepath as s_telepath
 
+# Avoid import cycle
```
Comment (vertexmc) [inline synapse/lib/cell.py]: I would remove this

File: synapse/lib/lmdb.py [hunk 6]
```diff
@@ -1,13 +1,11 @@
 import sys
 import struct
-import itertools
+
+import xxhash  # type: ignore
```
Comment (vertexmc) [inline synapse/lib/lmdb.py]: Does this type annotation need to be added to third party code or just modules w/o mypy stubs?

Comment (jnwatson) [reply synapse/lib/lmdb.py]: Just the modules without mypy stubs.

File: synapse/cryotank.py (hunk 1 above)
Comment (jnwatson) [reply synapse/cryotank.py]: typing is smart enough to infer the type from the constant.

File: synapse/cryotank.py (hunk 2 above)
Comment (jnwatson) [reply synapse/cryotank.py]: ok

**Pull request merged**